doc/configuration/Configuration.md

```diff
@@ -0,0 +1,319 @@
+### SMTP
+
+```"SMTPUsername": ""```
+*Obtain this credential from administrator setting up your email server.*
+
+```"SMTPPassword": ""```
+*Obtain this credential from administrator setting up your email server.*
+
+```"SMTPServer": ""```
+*Location of SMTP email server.*
+
+```"SMTPPort": ""```
+*Port of SMTP email server.*
+
+### Security
+
+```"ConnectionSecurity": ""```
+*"none": send email over an unsecure connection; "TLS": communication between Mattermost and your email server is encrypted; “STARTTLS”: attempts to upgrade an existing insecure connection to a secure connection using TLS.*
+
+```"InviteSalt": "bjlSR4QqkXFBr7TP4oDzlfZmcNuH9YoS"```
+*32-character (to be randomly generated via Admin Console) salt added to signing of email invites.*
+
+
+```"PasswordResetSalt": "vZ4DcKyVVRlKHHJpexcuXzojkE5PZ5eL"```
+*32-character (to be randomly generated via Admin Console) salt added to signing of password reset emails.*
+
+
+### File Settings
+___
+Settings to configure storage, appearance, and security of files and images.
+
+### File Storage
+
+```"DriverName": "local"```
+*System used for file storage. “local”: Files and images are stored on the local file system. “amazons3”: Files and images are stored on Amazon S3 based on the provided access key, bucket and region fields.*
+
+```"Directory": "./data/"```
+*Directory to which files are written. If blank, directory will be set to ./data/.*
+
+```"AmazonS3AccessKeyId": ""```
+*Obtain this credential from your Amazon EC2 administrator.*
+
+```"AmazonS3SecretAccessKey": ""```
+*Obtain this credential from your Amazon EC2 administrator.*
+
+```"AmazonS3Bucket": ""```
+*Name you selected for your S3 bucket in AWS.*
+
+```"AmazonS3Region": ""```
+*AWS region you selected for creating your S3 bucket.*
+
+### Image Settings
+
+```"ThumbnailWidth": 120```
+*Width of thumbnails generated from uploaded images. Updating this value changes how thumbnail images render in future, but does not change images created in the past.*
+
+```"ThumbnailHeight": 100```
+*Height of thumbnails generated from uploaded images. Updating this value changes how thumbnail images render in future, but does not change images created in the past.*
+
+```"PreviewWidth": 1024```
+*Maximum width of preview image. Updating this value changes how preview images render in future, but does not change images created in the past.*
+
+```"PreviewHeight": 0```
+*Maximum height of preview image ("0": Sets to auto-size). Updating this value changes how preview images render in future, but does not change images created in the past.*
+
+```"ProfileWidth": 128```
+*Width of a profile picture recommended when uploading from Account Settings. A users full size profile picture is shown when you hover over their name in a channel thread.*
+
+```"ProfileHeight": 128```
+*Height of a profile picture recommended when uploading from Account Settings. A users full size profile picture is shown when you hover over their name in a channel thread.*
```

Comment:
ProfileWidth and ProfileHeight aren't just the size recommended when uploading, we also resize any uploaded image to that size